compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/RewriteFft.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree/compiler/Dialect/LinalgExt/Transforms/Transforms.h"
+
+namespace mlir::iree_compiler::IREE::LinalgExt {
+
+std::tuple<LogicalResult, Value, Value> rewriteFft(Operation *op, Value operand,
+                                                   int64_t fftLength,
+                                                   PatternRewriter &rewriter) {
+
+  assert(!(fftLength & (fftLength - 1)) &&
```

Comment:
Use an llvm helper to check for powers of two: https://github.com/llvm/llvm-project/blob/7d1c661381d36018fd105f4ad4c2d6dc45e7288b/llvm/include/llvm/Support/MathExtras.h#L289-L298